src/main/java/apoc/convert/Json.java

```diff
@@ -82,20 +79,35 @@ public Object getJsonProperty(@Name("node") Node node, @Name("key") String key,@
             Iterator<PropertyContainer> it = path.iterator();
             while (it.hasNext()) {
                 Node n = (Node) it.next();
-                    Map<String, Object> nMap = maps.computeIfAbsent(n.getId(), (id) -> toMap(n));
+                Map<String, Object> nMap = maps.computeIfAbsent(n.getId(), (id) -> toMap(n));
                 if (it.hasNext()) {
                     Relationship r = (Relationship) it.next();
                     Node m = r.getOtherNode(n);
-                    Map<String, Object> mMap = maps.computeIfAbsent(m.getId(), (id) -> toMap(m));
-                    String typeName = r.getType().name();
-                    if (lowerCaseRels) typeName = typeName.toLowerCase();
-                    mMap = addRelProperties(mMap, typeName, r);
+                    String typeName = lowerCaseRels ? r.getType().name().toLowerCase() : r.getType().name();
                     // todo take direction into account and create collection into outgoing direction ??
                     // parent-[:HAS_CHILD]->(child) vs. (parent)<-[:PARENT_OF]-(child)
                     if (!nMap.containsKey(typeName)) nMap.put(typeName, new ArrayList<>(16));
-                    List list = (List) nMap.get(typeName);
-                    if (!list.contains(mMap))
```

Comment:
this is the only bit that has to be changed imho. the other changes are not relevant.
Also if the map is already in the `maps` datastructure doesn't it also imply that it's already in list.